src/filesystem.c

```diff
@@ -1653,15 +1654,23 @@ int FS_ReadLine( void *buffer, int len, fileHandle_t f ) {
 
 	buf = buffer;
         *buf = 0;
-	read = fgets (buf, len, fsh[f].handleFiles.file.o);
-
-
+	
+	//--- TODO: This should be updated to read until it properly encounters end of line characters.  
+	//---       Since fgets() will stop reading when it reads len-1 characters the end of line characters will be left for the next read.
+	//---       It should also not return the EOL characters.  This would eliminate the need to check for them.
+	
+	//read = fgets (buf, len, fsh[f].handleFiles.file.o);
+	read = fgets (line, sizeof(line), fsh[f].handleFiles.file.o);
```

Comment:
Reading maximum as many bytes as the length parameter allowed was the better way. You would have to handle this problem better in the calling function than editing this one.